builder/slide_and_fold_storage.cc

```diff
@@ -445,6 +255,9 @@ class slide_and_fold_storage : public node_mutator {
     merge_crop(bounds, op->bounds);
     if (bounds) {
       substitute_bounds(*bounds, current_buffer_bounds());
+      for (auto& b : *bounds) {
+        b = simplify(b);
```

Comment:
Isn't this simplification a bit heavy? Nested crops will do a lot of redundant simplification here...